### PR TITLE
test_infra: Move struct `Guest` and struct `GuestCommand` from tests

### PR DESCRIPTION
**Pull request by likebreath**
In this way, these structs can be reused for performance tests. 

This patch also requires  to provide `firmware path` to new `Guest`. 
In this way, we can remove hard-coded firmware names/paths from the
`test_infra` crate, and also have definition of constants for guest
images and firmware names defined along with the tests themselves.

Signed-off-by: Bo Chen <chen.bo@intel.com>

### CODE REVIEW

File: tests/integration.rs
Comment (likebreath) [inline tests/integration.rs:5678]: This is the only place where I found to be odd by following the original code path, that is by default we always invoke `fw_path(FwType::RustHypervisorFirmware)` when creating a new `Guest` even for `aarch64` (that uses its own hard-coded firmware name and ignores the provided `FwType`). 

If we are fine with changing the original code path, we can adjust to use `fn fw_path(#[cfg(target_arch = "x86_64")] fw_type: FwType) -> String` and just invoke `fw_path()` for `aarch64`. Any preference? @sboeuf

**Comment by likebreath** [comment]
Note: that there are several `vfio::` tests that is actually using the firmware, but are not exercised by the CI (as they are running on a periodic CI I think). We may want to manually run these tests before we merge this PR.

**Comment by sboeuf** [comment]
> Note: that there are several `vfio::` tests that is actually using the firmware, but are not exercised by the CI (as they are running on a periodic CI I think). We may want to manually run these tests before we merge this PR.

Yes you're right, I'll push an extra commit to this PR which will modify `Jenkinsfile` so that we get these tests running on the PR as well. Please make sure we don't merge the PR with this commit, this is only for testing purpose.

File: test_infra/src/lib.rs
Comment (sboeuf) [inline test_infra/src/lib.rs]: Do you need to move these in the first place?

File: tests/integration.rs
Comment (sboeuf) [inline tests/integration.rs:29]: Can't we fix this correctly? I mean by defining the import only for `x86_64`.

File: test_infra/src/lib.rs
Comment (sboeuf) [inline test_infra/src/lib.rs:673]: Ok by looking at the usage of `fw_path` from the `Guest` structure, I think we should remove it from the structure. The firmware can be directly created from `tests/integration.rs`, no need to be passed to the `Guest` structure, since 99% of the time, the value of `fw_path` is needed from within `tests/integration.rs`.
I think there's only one case that needs some special handling: `api_create_body()`. Simply add a new `fw_path` parameter to it.

File: tests/integration.rs
Comment (sboeuf) [reply tests/integration.rs:5678]: Following my suggestion about removing `fw_path` from `Guest`, I think that should simplify everything here since you'd simply invoke the local `fw_path()` for both architectures.

**Review by sboeuf — CHANGES_REQUESTED** [review]

File: test_infra/src/lib.rs
Comment (likebreath) [reply test_infra/src/lib.rs:673]: Good advise. Will update the PR accordingly.

File: test_infra/src/lib.rs
Comment (likebreath) [reply test_infra/src/lib.rs]: It was to make sure that every (the first) commit can be compiled correctly. If that's not something we want to ensure, I can merge the two "moves". Please let me know your thoughts.

Comment (sboeuf) [reply test_infra/src/lib.rs]: We want that, I just thought you didn't need to move these const.

Comment (likebreath) [reply test_infra/src/lib.rs]: Got it. Just realize that I can try to do the refactoring on the `fw_path()` first, and that should allow us to avoid the extra moves. :)

**Comment by likebreath** [comment]
@sboeuf  I believe all your previous comments are addressed. PTAL. Thank you.

**Review by sboeuf — APPROVED** [review]
Looks great!

**Pull request merged**
